File: backend/PortfolioApi/Extensions/StringSanitizer.cs
using System.Text.RegularExpressions;

namespace PortfolioAPI.Extensions;

public static partial class StringSanitizer
{
    public static string Clean(string? value, int maxLength)
    {
        if (string.IsNullOrWhiteSpace(value)) return string.Empty;
        var cleaned = HtmlTagRegex().Replace(value.Trim(), string.Empty);
        cleaned = cleaned.Replace("\0", string.Empty);
        return cleaned.Length <= maxLength ? cleaned : cleaned[..maxLength];
    }

    [GeneratedRegex("<.*?>", RegexOptions.Compiled)]
    private static partial Regex HtmlTagRegex();
}

File: backend/portfolio-api/src/main/java/com/nikhilwabale/portfolioapi/util/StringSanitizer.java
package com.nikhilwabale.portfolioapi.util;

import java.util.regex.Pattern;

public final class StringSanitizer {

    private static final Pattern HTML_TAG_PATTERN = Pattern.compile("<.*?>");

    private StringSanitizer() {
    }

    public static String clean(String value, int maxLength) {
        if (value == null || value.isBlank()) {
            return "";
        }

        String cleaned = HTML_TAG_PATTERN.matcher(value.trim()).replaceAll("");
        cleaned = cleaned.replace("\0", "");

        return cleaned.length() <= maxLength ? cleaned : cleaned.substring(0, maxLength);
    }
}
